test/controllers/pie_chart_controller_test.rb

```diff
@@ -0,0 +1,22 @@
+require "application_system_test_case"
+
+class PieChartControllerTest < ApplicationSystemTestCase
+
+  setup do
+    sign_in @user = users(:family_admin)
+    @account = accounts(:credit_card)
+  end
+
+  test "pie chart values is displayed correctly when cents are defined" do
+    assert_amount_balance '48550.80'
+  end
+
+  private
+  def assert_amount_balance(expected_value)
+    value_element = find('div.absolute.inset-0.w-full.text-center p span.text-xl') # Fetch the pie chart text
+    entire_text = value_element.find(:xpath, '..').text # Get full text of the <p> element
+    extracted_value = entire_text.strip.gsub(/[^\d.]/, '').sub(/^0+/, '') # Removing non-digit characters except the decimal point
+    assert_equal expected_value, extracted_value, "The pie chart value is not displayed correctly"
+  end
+
+end
```

Comment:
Great idea to add these tests!  A few ideas to make this test a bit more resilient:

- We should try and be less specific with our selectors here.  I'd go ahead and add an `id` or `data-id` on the specific element that you're looking to test so that when classes change in the future (which they likely will), this test will not break.
- I think if you want to write a system test like this, it should be in a file called `test/system/dashboard_test.rb`.  That said, since this is specific to the formatting of a monetary value which is provided by `value_groups_helper.rb`, it might be best to test this directly in `test/helpers/value_groups_helper_test.rb`

This is a pretty tricky one as it deals with the formatting of a currency.  Since we support multi-currency apps, it appears that the problem here may exist in the `pie_chart_controller.js`, which relies on the values to be in `USD`.

I think the change that we should consider making is removing or altering the `#currencyValue` method from `pie_chart_controller.js`.  This method relies on the decimal separator always being a `.` which will not apply for all currencies.  Switching this to `,` will then cause the opposite problem.  I'm not sure what the best solution here is, but my guess is that it would be easier to pass the Stimulus controller an already-formatted currency value so that we can leverage our robust collection of `Money::Currency` objects on the backend.

Reply:
I have made the logic in the backend to generate the value but i stick with a problem. The display is not really looking like time series chart is doing right now ( which is better ! ) 

Pie chart display : 
![image](https://github.com/maybe-finance/maybe/assets/39769262/75da0de5-a22d-4b64-ab31-924832210d54)

TimeSeries chart display : ( the title part )
![image](https://github.com/maybe-finance/maybe/assets/39769262/d9fe0d06-ace4-4b5a-b0e2-94f2b8b3f76c)

I can't find a way to display in gray the symbol, i don't known if using the ``format`` parameter of ``format_money`` function i use is a good approach to fix this problem..

I'm pushing my current changes, if you have some advice i will be happy !

Reply:
@AurelienConte if I'm following you correctly here, I believe this is expected behavior.

The time series graph will show _net worth_ (assets - liabilities) while the pie chart here is showing assets only.